tools/Config.mk

```diff
@@ -519,6 +519,8 @@ $(1)_$(2):
 
 endef
 
+ARCHDEFINES += ${shell $(DEFINE) "$(CC)" __NuttX__}
+
```

Comment:
is ARCHDEFINES appropriate for this purpose?

ARCHDEFINES sounds like arch-dependent flags to me. i feel `__NuttX__` is rather opposite.

Reply:
> is ARCHDEFINES appropriate for this purpose?

Not very appropriate, but there isn't more suitable variable.

> 
> ARCHDEFINES sounds like arch-dependent flags to me. i feel `__NuttX__` is rather opposite.

But, you can see many ARCHxxx defined in board's Make.defs aren't arch dependent. So I interpret ARCHxxx means the  predefined(either toolchain, arch, or system specific) macros. like ARCHINCLUDE defined here:
https://github.com/apache/incubator-nuttx/pull/2192/files#diff-16f6fa022b74dcae513f38d77f8979d829fdf1a19412e0ef4e8aecae53aed0d0L524
Do you think so?

Reply:
i guess the problem here is that the intention of ARCHxxx does not seem documented. i guess you can start documenting your interpretation.